### PR TITLE
Fix cpuinfo initialization failure in mlas test

### PR DESCRIPTION
**Pull request by chenfucn**
mlas test does not call environment initialization. so we need to manually initialize cpuinfo

### CODE REVIEW

File: onnxruntime/core/common/cpuid_info.cc [hunk 1]
```diff
@@ -63,57 +62,67 @@ static inline int XGETBV() {
 
 CPUIDInfo CPUIDInfo::instance_;
 
+std::once_flag cpuidinit_flag;
```
Comment (skottmckay) [inline onnxruntime/core/common/cpuid_info.cc:66]: Should these both be file scope statics or in an anonymous namespace?

File: onnxruntime/core/common/cpuid_info.cc [hunk 2]
```diff
@@ -11,15 +11,14 @@
...
-
 #if defined(_MSC_VER)
 #include <intrin.h>
 #elif defined(__GNUC__)
 #include <cpuid.h>
 #endif
 #endif
 
+#include <mutex>
 #include "core/common/cpuid_info.h"
```
Comment (skottmckay) [inline onnxruntime/core/common/cpuid_info.cc:22]: Can we use OrtMutex for consistency?

**Comment by snnn** [comment]
If you use C++ magic static, it could be a bit simpler.

Like this:

```c++
namespace {
struct KernelRegistryAndStatus {
 //It doesn't need be a std::shared_ptr, it can be any C++ type;
  std::shared_ptr<onnxruntime::KernelRegistry> kernel_registry = std::make_shared<onnxruntime::KernelRegistry>();
  onnxruntime::Status st;
};
}  // namespace



KernelRegistryAndStatus GetCpuKernelRegistry() {
  KernelRegistryAndStatus ret;
   
  //Do the init here. Fill out kernel_registry  and set the status.

  return ret;
}

//The returned value doesn't need be a std::shared_ptr, it can be any C++ type;
std::shared_ptr<KernelRegistry> CPUExecutionProvider::GetKernelRegistry() const {
  static KernelRegistryAndStatus k = GetCpuKernelRegistry();
  //throw if the registry failed to initialize
  ORT_THROW_IF_ERROR(k.st);
  return k.kernel_registry;
}

```

**Comment by chenfucn** [comment]
Thanks guys. I guess I should have used the original solution for initialization before my first change.

**Review by skottmckay — APPROVED** [review]
:shipit:

File: onnxruntime/core/common/cpuid_info.cc [hunk 3]
```diff
@@ -64,56 +63,54 @@ static inline int XGETBV() {
...
+CPUIDInfo::CPUIDInfo() {
 #ifdef CPUINFO_INCLUDED
-  if (!cpuinfo_initialize()) {
-    // Unfortunately we can not capture cpuinfo log!!
-    return ORT_MAKE_STATUS(SYSTEM, FAIL, "Failed to initialize cpuinfo");
-  }
+    if (!cpuinfo_initialize()) {
+      // Unfortunately we can not capture cpuinfo log!!
+      ORT_THROW("Failed to initialize CPU info.S");
+    }
```
Comment (skottmckay) [inline onnxruntime/core/common/cpuid_info.cc:71]: nit: is the trailing 'S' intentional?

Comment (chenfucn) [reply onnxruntime/core/common/cpuid_info.cc:71]: that was evidence of my stupidity and now that evidence has been destroyed.

**Review by snnn — APPROVED** [review]

**Pull request merged**